src/slashcommands/Birthday.ts

```diff
@@ -167,34 +196,67 @@ export class Birthday implements SlashCommand {
 			});
 			return;
 		}
-		if (interaction.options.getSubcommand() == 'channel') {
-			//TODO: Permission Check
-			//TODO: verify user is an admin
+		if (interaction.options.getSubcommand() == 'config') {
+			if (!(interaction.channel instanceof TextChannel)) {
+				interaction.reply('Command must be used in a server');
+				return;
+			}
+			let member = interaction.member as GuildMember;
+			if (!member.permissionsIn(interaction.channel!).has('ADMINISTRATOR')) {
+				interaction.reply({
+					content:
+						'This command is only for people with Administrator permissions',
+					ephemeral: true,
+				});
+				return;
+			}
 
-			//ensure that the enmap has stored the guild and brings in the JSON. If not create it and give it an empty JSON.
-			var guildChannel = birthdayChannels.ensure(
-				`${interaction.guild!.id}`,
-				''
-			);
-			//store the date of birth in numerical form
+			var guildChannel = bdayChannels.ensure(interaction.guild!.id, '');
```

Comment:
Another case of getting a variable from an enmap using ensure when it's not necessary